pyanaconda/modules/subscription/subscription.py

```diff
@@ -510,3 +515,41 @@ def rhsm_observer(self):
         :rtype: RHSMObserver instance
         """
         return self._rhsm_observer
+
+    @property
+    def rhsm_config_defaults(self):
+        """Return RHSM config default values.
+
+        We need to have these available in case the user decides
+        to return to default values from a custom value at
+        runtime.
+
+        NOTE: The property is lazy evaluated, the first attempt
+        to read it fetches the full config dict from RHSM and
+        subsequent reads are then served from cache. Due to this
+        it is important no to set RHSM configuration values
+        first "warming up" this property by reading it or else
+        it might return non-default (Anaconda overwritten) data.
+
+        :return : dictionary of default RHSM configuration values
+        :rtype: dict
+        """
+        if self._rhsm_config_defaults is None:
+            # config defaults cache not yet populated, do it now
+            proxy = self.rhsm_observer.get_proxy(RHSM_CONFIG)
+            self._rhsm_config_defaults = proxy.GetAll("")
+        return self._rhsm_config_defaults
+
+    def set_rhsm_config_with_task(self):
+        """Set RHSM config values based on current subscription request.
+
+        :return: a DBus path of an installation task
+        """
+        # NOTE: we access self._subscription_request directly
+        #       to avoid the sensitive data clearing happening
+        #       in the subscription_request property getter
```

Comment:
This note can be solved by new read-only getter with self describing name. It would take less lines than this explanation.

Reply:
Yeah, I like that suggestion - it makes more sense for a method to have side effects like that rather than for a property. Lets do that. :)